Helper/Data.php

```
@@ -254,6 +283,271 @@ private function sendApiRequest(array $payload)
+  /**
+   * Deliver a queued event over HTTP.
+   *
+   * Loads the stored payload and performs the HMAC + cURL exchange.
+   * Does NOT touch any state column — that is the caller's responsibility.
+   *
+   * @param int    $eventId
+   * @param string $workerId  (unused by HTTP layer, reserved for future tracing)
+   * @return string|null  null on HTTP 200; otherwise a truncated error message
+   *                      that the caller should persist via recordFailedAttempt.
+   */
+  public function deliverEvent(int $eventId, string $workerId): ?string
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$workerId'. (undefined)

(UnusedFormalParameter)